controllers/vmware/vspherecluster_reconciler_test.go

```diff
@@ -130,32 +133,137 @@ var _ = Describe("Cluster Controller Tests", func() {
 	})
 
 	Context("Test getFailureDomains", func() {
```

Comment:
Let's remove this whole context and have a normal go style table-test instead. Let me know if you need help or guidance.

I can also create a PR on top of yours if you're busy or prefer that :-)

Reply:
I assume this is not a small change :) please feel free to go ahead to refine it, I'll continue to address other review comments and the local testing with supervisor.